### DISCUSSION TITLE
How to understand the PQ_OETF function correctly?

### DISCUSSION THREAD

**Discussion by sherlockedlee** [colour-science/colour#952] [Q&A]
I am a little confused about the input of the PQ_OETF. In the camera pipeline, when input is a picture(10bit .raw for instance), how to get the absolute cv-nit according to the value RGB？ What does "the signal determined by scene light and scaled by camera exposure" mean? Is that the value RGB in the raw file? If the content of the raw file is BT.2020, how to use the OOTF correctly like BT.709? How to calculate the constant 59.5208?
`100 * eotf_BT1886(oetf_BT709(59.5208 * E))`

**Comment by thomasmansencal** [comment]
Hi @sherlockedlee,

> In the camera pipeline, when input is a picture(10bit .raw for instance), how to get the absolute cv-nit according to the value RGB?

You would need to first know the camera calibration constant _C_ for the camera and then use the exposure equation to compute the absolute illuminance incident to the sensor plane: https://en.wikipedia.org/wiki/Exposure_value#Relationship_of_EV_to_lighting_conditions, https://github.com/wetadigital/physlight/blob/master/physlight_imaging.ipynb

Then you also need to know how the data is encoded in the 10-bit container because it will certainly not be linearly encoded, e.g. Logarithmic encoding or alike.

> What does "the signal determined by scene light and scaled by camera exposure" mean? Is that the value RGB in the raw file?

Yes!

>  If the content of the raw file is BT.2020, how to use the OOTF correctly like BT.709? 

BT.2020 here probably refers to the encoding _gamut_ not the _transfer function_, so the OOTF will apply independently of the underlying gamut.

> How to calculate the constant 59.5208?

I'm not sure to understand, are you asking how it was derived?

**Reply by sherlockedlee** [reply]
Hi @KelSolaar ,
Thanks so much for your answer. I am a newcomer to color science and can not fully understand your answer. In fact, I am confused about the meaning of "absolute luminance" of PQ_OETF input which is mentioned in many materials.

> What does "the signal determined by scene light and scaled by camera exposure" mean? Is that the value RGB in the raw file?
> Yes!

1. I notice the input of curve PQ_OETF has the uint--nit while RGB values in raw files don't. Should I estimate the real scene luminance from the RGB value in the raw file first like many HDRI methods do(which is no need according to your code) and then perform the PQ_OETF function?

> How to calculate the constant 59.5208?
> I'm not sure to understand, are you asking how it was derived?

2. I understand "OOTF will apply independently of the underlying gamut". I notice the PQ process can be bt2100_oetf(ootf_2100(x)), and OOTF_2100 should be rec_1886_eotf(rec_709_oetf(59.5208 * x)). Because the OETF of Rec.709 is not very same as BT.2020_precise, I wonder how to decide the constant 59.5208 which is hyperparameter and different from the original oetf_709. And through this transformation, do we map the highest luminance in SDR(100nit) directly to the HDR(like 10000nit)? BTW, I have calculated the constant 59.5208 by solving the equation mapping the FD from 100 to 10000 using Matlab but not sure whether the process is correct.
